spec/lib/trs/params/professional_status_spec.rb

```diff
@@ -0,0 +1,209 @@
+
+          before do
+            allow(::CodeSets::Trs).to receive(:training_status).with(trainee.state, trainee.training_route).and_return("Withdrawn")
+          end
+
+          it "sets the status to Withdrawn" do
+            expect(subject["status"]).to eq("Withdrawn")
+          end
+        end
+
+        context "trainee is recommended for award" do
+          let(:trainee) { create(:trainee, :completed, :recommended_for_award) }
+
+          before do
+            allow(::CodeSets::Trs).to receive(:training_status).with(trainee.state, trainee.training_route).and_return("Approved")
+          end
+
+          it "sets the status to Approved" do
+            expect(subject["status"]).to eq("Approved")
+          end
+        end
+
+        context "trainee is awarded" do
+          let(:trainee) { create(:trainee, :completed, :awarded) }
+
+          before do
+            allow(::CodeSets::Trs).to receive(:training_status).with(trainee.state, trainee.training_route).and_return("Awarded")
+          end
+
+          it "sets the status to Awarded" do
+            expect(subject["status"]).to eq("Awarded")
+            expect(subject["awardedDate"]).to eq(trainee.awarded_at.to_date.iso8601)
+          end
+        end
+
+        context "trainee is in training (non-assessment only)" do
+          let(:trainee) { create(:trainee, :completed, :provider_led_postgrad, :trn_received) }
+
+          before do
+            allow(::CodeSets::Trs).to receive(:training_status).with(trainee.state, trainee.training_route).and_return("InTraining")
+          end
+
+          it "sets the status to InTraining" do
+            expect(subject["status"]).to eq("InTraining")
+          end
+
+          it "sets the correct route type ID" do
+            expect(subject["routeTypeId"]).to eq("97497716-5ac5-49aa-a444-27fa3e2c152a")
+          end
+        end
+
+        context "trainee is iQTS" do
+          let(:trainee) { create(:trainee, :completed, :iqts, iqts_country: "Ireland") }
+          let(:country_ref) { "IE" }
+          let(:professional_status) { instance_double(described_class) }
+          let(:params_result) { { "trainingCountryReference" => country_ref, "routeTypeId" => "d0b60864-ab1c-4d49-a5c2-ff4bd9872ee1" } }
+
+          before do
+            allow(::CodeSets::Trs).to receive(:training_status).with(trainee.state, trainee.training_route).and_return("UnderAssessment")
+            allow(described_class).to receive(:new).with(trainee:).and_return(professional_status)
+            allow(professional_status).to receive(:params).and_return(params_result)
+          end
+
+          it "sets the route type and training country code appropriately" do
+            expect(professional_status.params).to include({
+              "trainingCountryReference" => country_ref,
+              "routeTypeId" => "d0b60864-ab1c-4d49-a5c2-ff4bd9872ee1",
+            })
+          end
+        end
+      end
```

Comment:
This is over engineered

Reply:
re-writing this and the rest of this spec